### PR TITLE
[BEAM-5731] Only accept key operator in Python 3.

### PR DESCRIPTION
**Pull request by robertwb**
This is in line with Python's change to its sorting operations.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 1]
```diff
@@ -1875,7 +1875,7 @@ def test_per_key_pipeline_checking_violated(self):
       (self.p
        | beam.Create(range(100)).with_output_types(int)
        | 'Num + 1' >> beam.Map(lambda x: x + 1).with_output_types(int)
-       | 'TopMod' >> combine.Top.PerKey(1, lambda a, b: a < b))
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Should we leave some of these tests for the remaining py2 code?

File: sdks/python/apache_beam/transforms/combiners_test.py [hunk 2]
```diff
@@ -138,6 +152,8 @@ def test_top_key(self):
         ['aa', 'bbb', 'c', 'dddd'] | combine.Top.Of(3, key=len, reverse=True),
         [['c', 'aa', 'bbb']])
 
+  @unittest.skipIf(sys.version_info[0] > 1, 'deprecated comparator')
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners_test.py]: Will `sys.version_info[0] > 1` ever evaluate to True? Should we remove this entirely?

Comment (tvalentyn) [reply sdks/python/apache_beam/transforms/combiners_test.py]: *evaluate to False. Did you mean > 2?

Comment (aaltay) [reply sdks/python/apache_beam/transforms/combiners_test.py]: I believe you are right, updated the two uses of it here.

File: sdks/python/apache_beam/transforms/ptransform_test.py (hunk 1 above)
Comment (aaltay) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: Merging this as is. @robertwb please address this comment in a follow up.

Comment (robertwb) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: The choice of compare function here does not influence the type inference (which is the point of this test). Incidentally, these were also the default (natural ordering) so entirely redundant.

File: sdks/python/apache_beam/transforms/combiners_test.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/transforms/combiners_test.py]: Oh, thanks. I was just testing this on Py2.

**Pull request merged**